src/Chapter2.hs

```diff
@@ -832,7 +873,16 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n aList
+  | n < 0 || length aList == 0 = []
```

Comment:
Here the same note about length.

Also, I see that you already have it as the variable in `where`, so you can reuse it in here 🙂